internal/services/appservice/linux_function_app_resource.go

```diff
@@ -547,6 +575,43 @@ func (r LinuxFunctionAppResource) Read() sdk.ResourceFunc {
 				return fmt.Errorf("reading App Settings for Linux %s: %+v", id, err)
 			}
 
+			var storageAccountName, storageFileShare string
+			if sendContentSettings || helpers.PlanIsConsumption(planSku) {
+				if appSettingsResp.Properties != nil {
+					if appSettingsResp.Properties["WEBSITE_CONTENTAZUREFILECONNECTIONSTRING"] != nil {
+						saConnectionString := strings.Split(*appSettingsResp.Properties["WEBSITE_CONTENTAZUREFILECONNECTIONSTRING"], ";")
+						for _, part := range saConnectionString {
+							if strings.Contains(part, "AccountName") {
+								storageAccountName = strings.TrimPrefix(part, "AccountName=")
+								break
+							}
+						}
+					}
+					if appSettingsResp.Properties["WEBSITE_CONTENTSHARE"] != nil {
+						storageFileShare = *appSettingsResp.Properties["WEBSITE_CONTENTSHARE"]
+					}
+					saClient := metadata.Client.Storage
+					account, err := saClient.FindAccount(ctx, storageAccountName)
+					if err != nil {
+						return fmt.Errorf("retrieving Account %q for Share %q: %s", storageAccountName, storageFileShare, err)
+					}
+					if account == nil {
+						return fmt.Errorf("unable to locate Storage Account %s", storageAccountName)
+					}
+					saFileShareClient, err := metadata.Client.Storage.FileSharesClient(ctx, *account)
+					if err != nil {
+						return fmt.Errorf("building File Share Client: %s", err)
+					}
+					exists, err := saFileShareClient.Exists(ctx, account.ResourceGroup, storageAccountName, storageFileShare)
+					if err != nil {
+						return fmt.Errorf("checking for existence of Storage Share %q (Account %q) : %+v", storageFileShare, storageAccountName, err)
+					}
+					if exists == nil {
+						return fmt.Errorf("the storage file share %s for function app %s does not exists, please specify an existing one", storageFileShare, id.ID())
+					}
+				}
+			}
+
```

Comment:
As above, none of this should be in a read function, we should be setting received data back into state only, not performing configuration checks on external resources that we're being told about.

Reply:
the same to the reply in the windows function app